wpcom-legacy-redirector.php

```diff
@@ -170,12 +171,112 @@ static function get_redirect_post_id( $url ) {
 
 		$redirect_post_id = $wpdb->get_var( $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE post_type = %s AND post_name = %s LIMIT 1", self::POST_TYPE, $url_hash ) );
 
-		if ( ! $redirect_post_id )
+		if ( ! $redirect_post_id ) {
 			$redirect_post_id = 0;
+		}
 
 		return $redirect_post_id;
 	}
 
+	/**
+	 * Validate the redirect should work normally.
+	 *
+	 * @param array $redirect Redirect array.
+	 * @param array $post_types Array of publicly accessible post types.
+	 * @return bool|string True if the redirect passes validation. Array containing error notice if redirect fails validation.
+	 */
+	static function validate_redirect( $redirect, $post_types ) {
```

Comment:
Should we separate this out into separate methods: one for url and one for post?